### PR TITLE
[python] Fix database error handler about no datasource name

### LINKED ISSUE #7616: [Bug] [python] Class database raise error NPE when database name not exists
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When database source name not exists, would raise NPE without detail info, it not user friendly.

### What you expected to happen

should remind user that datasouece not exists

### How to reproduce

run `python ./example/task_datax_example.py`

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongjiajie**
The original code show too many unrelated log when database name
do not exists. BTW, it would continue the code when error miss.
This patch is try to fix it.

close: #7616

### CODE REVIEW

**Comment by zhongjiajie** [comment]
@devosend PTAL, if you have time, thanks.

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/database.py [hunk 1]
```diff
@@ -52,5 +55,10 @@ def get_database_info(self, name) -> Dict:
             return self._database
         else:
             gateway = launch_gateway()
-            self._database = gateway.entry_point.getDatasourceInfo(name)
+            try:
+                self._database = gateway.entry_point.getDatasourceInfo(name)
+            # Handler database source do not exists error, for now we just terminate the process.
+            except Py4JJavaError:
```
Comment (devosend) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/database.py]: Except error is not just a situation. Should we use `Py4JJavaError.java_exception` to display clear information?

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/database.py]: It seems that `Py4JJavaError` only has one single subclass `java_exception`. https://www.py4j.org/py4j_java_protocol.html#py4j.protocol.Py4JJavaError. And it more like we could get detail java gateway information from client-side. But I would try it in my local environment.

**Comment by zhongjiajie** [comment]
I here allow it raise error base on py4j jave gateway server track stack, and I directly using chaining error message for more easy location the error. You could see in keyword `During handling of the above exception, another exception occurred`

Here it is an example 

```log
Traceback (most recent call last):
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/database.py", line 59, in get_database_info
    self._database = gateway.entry_point.getDatasourceInfo(name)
  File "/Users/zhongjiajie/.virtualenvs/pydolphinschduler/lib/python3.8/site-packages/py4j/java_gateway.py", line 1309, in __call__
    return_value = get_return_value(
  File "/Users/zhongjiajie/.virtualenvs/pydolphinschduler/lib/python3.8/site-packages/py4j/protocol.py", line 326, in get_return_value
    raise Py4JJavaError(
py4j.protocol.Py4JJavaError: An error occurred while calling t.getDatasourceInfo.
: java.lang.IllegalArgumentException: Can not find any datasource by name first_mysql
	at org.apache.dolphinscheduler.server.PythonGatewayServer.getDatasourceInfo(PythonGatewayServer.java:398)
	at sun.reflect.GeneratedMethodAccessor187.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/examples/task_datax_example.py", line 94, in <module>
    pd.run()
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py", line 325, in run
    self.submit()
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py", line 377, in submit
    json.dumps(self.task_definition_json),
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py", line 215, in task_definition_json
    return [task.get_define() for task in self.tasks.values()]
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py", line 215, in <listcomp>
    return [task.get_define() for task in self.tasks.values()]
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/base.py", line 69, in get_define
    content = self.get_define_custom(camel_attr, self._DEFINE_ATTR)
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/base.py", line 56, in get_define_custom
    val = getattr(self, attr, None)
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/datax.py", line 116, in task_params
    datasource = Database(self.datasource_name, "dsType", "dataSource")
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/database.py", line 39, in __init__
    self[type_key] = self.database_type
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/database.py", line 45, in database_type
    return self.get_database_info(self.database_name).get("type")
  File "/Users/zhongjiajie/Documents/program/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/database.py", line 62, in get_database_info
    raise PyDSParamException(str(ex.java_exception))
pydolphinscheduler.exceptions.PyDSParamException: java.lang.IllegalArgumentException: Can not find any datasource by name first_mysql

Process finished with exit code 1
```

**Comment by zhongjiajie** [comment]
PTAL @devosend

**Comment by devosend** [comment]
> PTAL @devosend

LGTM @zhongjiajie

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by zhongjiajie** [comment]
Thanks

**Pull request merged**